iana/ip.go

```diff
@@ -0,0 +1,179 @@
+package iana
+
+import (
+	"bytes"
+	"encoding/csv"
+	"errors"
+	"fmt"
+	"io"
+	"net/netip"
+	"regexp"
+	"slices"
+	"strings"
+
+	_ "embed"
+)
+
+type reservedPrefix struct {
+	// addressFamily values are defined in:
+	// https://www.iana.org/assignments/address-family-numbers/address-family-numbers.xhtml
+	addressFamily uint16
```

Comment:
On the one hand, using these integers that have been so kindly predefined for you makes sense.

On the other hand, these integers appear to _only_ be used for logging purposes: I see a lot of "failed to parse reserved address registry %d", and no other uses. So since this is going to be used only as a human-readable display string, it should be a human-readable "IPv4" or "IPv6" instead, imo.